controllers/provider-openstack/pkg/webhook/controlplanebackup/ensurer.go

```diff
@@ -138,6 +138,15 @@ func (e *ensurer) getBackupRestoreContainer(name string, cluster *extensionscont
 						},
 					},
 				},
+				{
+					Name: "OS_DOMAIN_ID",
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							Key:                  openstack.DomainID,
```

Comment:
OK, so adding these env variables always doesn't harm, right?
However, what happens if this key does not exist in the referenced secret? Will the pod come up anyways and then the value of the env variable will be `""`?

Reply:
The values are non pointer strings so yes, they will become "". Setting those envs doesn't harm, as the gophercloud authenticator reads and evaluates the environment variables correctly. https://github.com/gophercloud/gophercloud/blob/master/openstack/auth_env.go#L36

Reply:
Hm, actually, the pod won't come up if the referenced keys don't exist in the secret. Just ran into this problem (cc @swapnilgm).

Reply:
Yes. It shows "ContainerConfigError".

Reply:
Ok, then I need to make this conditional.

Reply:
So this part need to be completely refactored. I added here secrets file and dir handling to the etcd-backup pod. That way we can just dump the correct secrets into the pod without having to fiddle around with the pod spec validation. The OpenStack authentication mechanics are then handling the authentication correctly. https://github.com/gardener/gardener-extensions/pull/287

Reply:
@swapnilgm I just updated the PR with the correct handling of `cloud.yaml` in the etcd-backup pod spec.